src/components/carousel/carousel.component.ts

```diff
@@ -201,34 +227,36 @@ export default class SlCarousel extends ShoelaceElement {
 
       // Scrolls to the original slide without animating, so the user won't notice that the position has changed
       this.goToSlide(clonePosition, 'auto');
-
-      return;
+    } else if (firstIntersecting) {
+      // Update the current index based on the first visible slide
+      const slideIndex = slides.indexOf(firstIntersecting.target as SlCarouselItem);
+      // Set the index to the first "snappable" slide
+      this.activeSlide = Math.ceil(slideIndex / this.slidesPerMove) * this.slidesPerMove;
     }
+  }
 
-    // Activate the first intersecting slide
-    if (firstIntersecting) {
-      this.activeSlide = slides.indexOf(firstIntersecting.target as SlCarouselItem);
-    }
+  private isCarouselItem(node: HTMLElement): node is SlCarouselItem {
+    return node.tagName.toLowerCase() === 'sl-carousel-item';
```

Comment:
Moved this here to avoid importing the carousel item. I'm not sure though if this is the right approach, thinking also on future work for scoped custome elements.

Reply:
This is good for now. We'll have a lot of similar comparisons we'll need to address if/when scoped custom elements land.